lua/neogit/popups/commit.lua

```diff
@@ -6,11 +6,26 @@ local a = require('neogit.async')
 local uv = require('neogit.async.uv')
 local split = require('neogit.lib.util').split
 
-local COMMIT_FILE = '.git/NEOGIT_COMMIT_EDITMSG'
+local find_root = function()
+  local base_pwd = vim.fn.getcwd()
+  local target = string.byte("/")
+  for idx = #base_pwd, 1, -1 do
+    if base_pwd:byte(idx) == target then
+      local gitpath = base_pwd:sub(1, idx) .. ".git"
+      if vim.fn.isdirectory(gitpath) > 0 then
+        return gitpath
+      end
+    end
+  end
+  return ".git"
+end
```

Comment:
Just wondering, since separate worktrees for git do not need to be in a subdirectory of the repository, does this actually solve #90?
> So for example the main local repo is at /home/user/projects/foo and a worktree of it is in /home/user/projects/foo-prs, then the .git filesystem entry in the latter repo is a file that includes /home/user/projects/foo/.git/worktree/foo-prs.

Haven't tried it, but this doesn't look like it would be solved by above function.

Also, we already have a function to get the repository root here:
https://github.com/TimUntersberger/neogit/blob/2f504fcf734ddb6b281a93e01782ecb0dae0774e/lua/neogit/lib/git/cli.lua#L185-L187
I wonder if this one has the same problems